qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/FieldCapsIT.java

```diff
@@ -252,4 +262,46 @@ public void testAllIndicesWithIndexFilter() throws Exception {
         assertThat(resp.getField("blue_field").keySet(), contains("keyword"));
         assertTrue(resp.getField("blue_field").get("keyword").isSearchable());
     }
+
+    @SuppressWarnings("unchecked")
+    private RestClient getUpgradedNodeClient() throws IOException {
+        for (HttpHost host : getClusterHosts()) {
```

Comment:
you can plug in a NodeSelector to the RestClient to have it only point to nodes with specific characteristics. Version can be a factor.

Reply:
Frustratingly that doesn't help here, because the RestClientBuilder doesn't know the version of its Nodes up-front, and so we can't get the version in the NodeSelector.

Reply:
ok for my understanding, maybe the problem is that the existing client does not point to the upgraded nodes? I guess it depends on how it is initialized. The point of node selector is to influence node selection when making each request, but then all nodes need to be added to the client rotation. I see that the client gets the nodes from tests.rest.cluster set on build.gradle. Is there a chance that not all nodes are included?

Reply:
All the nodes are included; the problem is that the `RestClientBuilder.builder(HttpHost hosts...)` that gets called here then builds its Nodes using plain `Node::new`, which doesn't set any Node metadata (including the version).

Reply:
got it. Only the sniffer sets these automatically as it has knowledge about the metadata, but you can set metadata up yourself by using the builder(Node ... nodes) method. That makes this change not so useful and along the lines of what you already have though. Have a look if it makes sense to make the change or not?